src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { forwardRef, useRef } from "react";
+import cx from "classnames";
+import { useMergeRefs } from "../../hooks";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import { getTestId } from "../../tests/test-ids-utils";
+import { ComponentDefaultTestId } from "../../tests/constants";
+import styles from "./Badge.module.scss";
+import { Direction } from "./BadgeConstants";
+import Indicator, { IndicatorProps } from "../Indicator/Indicator";
+import Counter, { CounterProps } from "../Counter/Counter";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+
+type Position = {
+  vertical: Direction.TOP | Direction.BOTTOM;
+  horizontal: Direction.LEFT | Direction.RIGHT;
+};
+
+type BadgeOptions = typeof Indicator | typeof Counter;
+
+export interface BadgeProps extends VibeComponentProps, React.PropsWithChildren {
+  badgeComponent?: BadgeOptions;
+  position?: Position;
+  gap?: boolean;
+  circular?: boolean;
+  show?: boolean;
+  badgeComponentProps?: CounterProps | IndicatorProps;
+}
+
+const Badge: VibeComponent<BadgeProps> & {
+  directions?: typeof Direction;
+} = forwardRef(
+  (
+    {
+      badgeComponent: BadgeComponent = Indicator,
+      position = { vertical: Direction.TOP, horizontal: Direction.RIGHT } as Position,
+      gap = false,
+      circular = false,
+      show = true,
+      badgeComponentProps,
+      className,
+      id,
+      "data-testid": dataTestId,
+      children
+    },
+    ref
+  ) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const badgeClassNames = cx(
+      styles.badge,
+      getStyle(styles, camelCase(`${position.vertical}-${position.horizontal}`)),
+      { [styles.gap]: gap, [styles.circular]: circular, [styles.show]: show },
+      className
+    );
+
+    return (
+      <div ref={mergedRef} className={cx(styles.badgeWrapper)} id={id}>
```

Comment:
the class name purpose is always positioning and layouting the component and therefore it should be on the wrapper element.

Reply:
Also I think it will make more sense that the data test id will be on the wrapper element but let's talk about it